src/internal/observable/fromEvent.ts

```diff
@@ -70,11 +70,11 @@ export interface AddEventListenerOptions extends EventListenerOptions {
 
 export function fromEvent<T>(target: FromEventTarget<T>, eventName: string): Observable<T>;
 /** @deprecated resultSelector no longer supported, pipe to map instead */
-export function fromEvent<T>(target: FromEventTarget<T>, eventName: string, resultSelector?: (...args: any[]) => T): Observable<T>;
-export function fromEvent<T>(target: FromEventTarget<T>, eventName: string, options?: EventListenerOptions): Observable<T>;
+export function fromEvent<T>(target: FromEventTarget<any>, eventName: string, resultSelector: (...args: any[]) => T): Observable<T>;
```

Comment:
This seems to make sense. We're loosing up the type here, because really it's the `T` that matters. However, part of me feels like we should undeprecate this signature (as discussed in #5824), and in that case, we should try to get the types correct that are being passed to the result selector. I suppose we can do that in another pass though.

Reply:
Yeah. That's something that can be done after it's split into separate signatures. ATM, `FromEventTarget` makes the function look simpler than it really is and the `T` is really only applicable to _some_ of the signatures.